extension/apple/ExecuTorch/Exported/ExecuTorchModule.mm

```diff
@@ -331,12 +331,21 @@ - (nullable ExecuTorchMethodMetadata *)methodMetadata:(NSString *)methodName
 - (nullable NSArray<ExecuTorchValue *> *)executeMethod:(NSString *)methodName
                                             withInputs:(NSArray<ExecuTorchValue *> *)values
                                                  error:(NSError **)error {
-  std::vector<EValue> inputs;
-  inputs.reserve(values.count);
-  for (ExecuTorchValue *value in values) {
-    inputs.push_back(toEValue(value));
+  const char *methodNameString = methodName.UTF8String;
+  __block auto errorCode = Error::Ok;
+  [values enumerateObjectsUsingBlock:^(ExecuTorchValue *value, NSUInteger index, BOOL *stop) {
+    errorCode = _module->set_input(methodNameString, toEValue(value), index);
+    if (errorCode != Error::Ok) {
+      *stop = YES;
+    }
+  }];
+  if (errorCode != Error::Ok) {
+    if (error) {
+      *error = ExecuTorchErrorWithCode((ExecuTorchErrorCode)errorCode);
+    }
+    return nil;
   }
-  const auto result = _module->execute(methodName.UTF8String, inputs);
+  const auto result = _module->execute(methodNameString);
```

Comment:
For my understanding, I can still call execute again right after this without `set_input` again right? I.e. the input_set state doesn't get reset after each execute, right?